integrations/extensions/starter-kits/ibm-rpa/README.md

```diff
@@ -0,0 +1,107 @@
+
+## Integration Setup
+
+### Prerequisites
+
+To complete an end-to-end setup of this integration between Watson Assistant and IBM RPA you will need to provision accounts for all of the following:
+
+ - Watson Assistant (a trial account)
+ - IBM RPA
+ - IBM Cloud Functions (or some equivalent)
+
+### Setup RPA
+
+Go [here](https://www.ibm.com/docs/en/rpa/21.0?topic=automation-deploying-scripts) first for details on building and deploying RPA scripts.
+
+These instructions outline the basic steps: 
+
+1. Add credentials to your local machine so it can be unlocked.
+2. Build your script and publish (note the input and output variables as these will be used through out the setup of this integration).
+3. Define a system queue to use on the server.
+4. Workflow -> Processes -> Create Process
+    - Ensure variables match script (step 2) variables.
+
+Now ensure you can call that script via the rest api (Asynch/Synch). Note that this will require credentials.
+
+### Setup Cloud Functions
+
+First get the necessary parameters needed by the cloud functions:
+
+1. Call IBM RPA Tenants API from Postman or CURL and get the WorkspaceID.
+2. Login to IBM RPA Web Client, Go to processes that you created and get the process ID in the URL. For example: https://us1app.wdgautomation.com/#/en-US/workflows/processes/2a88c404-926c-44fe-b1b9-66c1620cc2ca) -> 2a88c404-926c-44fe-b1b9-66c1620cc2ca
+3. For security purposes go to the namespace section of cloud function and get the GUID.
+4. Know the API URL for your RPA Tenant (us1/BR2,UK1, etc).
+5. Know your RPA username and passowrd.
+
+Next create a cloud function used to insert an item in the RPA queue:
+
+1. Open the cfInsert.js template file included with this starter kit in the text editor of your choice.
+2. Write your personal values from above into the top of the cfInsert.js template script.
+3. Dynamic Parameters include: 
+    - RPA Username 
+    - RPA Password 
+    - appGUID 
+    - apiURL
+    - workspaceAPIID
+    - processID
+4. Build dynamic logic to insert the RPA expected variables (must match RPA script variables defined above). This involves modifying the script file variable names to match your RPA script (e.g., rpaInputParam1).
+5. Enable endpoint as a Web Action
+
+Next create the cloud function for reading status of an item in the queue:
+1. Open the cfGetStatus.js template file included with this starter kit in the text editor of your choice.
+2. Write your personal values from above into the top of the cfGetStatus.js template script.
+3. Dynamic Parameters include: 
+    - RPA Username
+    - RPA Password
+    - appGUID
+    - apiURL
+    - workspaceAPIID
+    - processID
+4. Enable endpoint as a Web Action
+
+Next update the OpenAPI spec template included with this starter kit
+1. Open the RPAAsynchExtensionAPISpec.json template file included with this starter kit in the text editor of your choice.
+2. Copy and paste over necessary lines in the template with the following:
```

Comment:
it would have been much nicer if your open api spec server would have been structured as follows:
```
"url": "https://{{region}}.functions.cloud.ibm.com/api/v1/web/{{cloud_function_guid}}/default"
   
```

Then you can put here a screenshot from WA on how one can just add the `region` and `cloud function identifier`

Reply:
done.